### ISSUE TITLE
Some views seem to still be rendered twice on the client

### ISSUE THREAD

**Issue by ingiulio** [rendrjs/rendr#461]
While playing with our app, I noticed the following into the normal flow of _shared/base/view.js_:

* The app is first started on the client, and `attachChildViews` is called from the main app view, so when we get to `BaseView.attach`, the `list` param will include all the views included in the page, i.e something like:

```
[
  childView1, 
  childView2,
  childView3,
  childViewOfChildView1
]
```

* When we get in the body of `async.map` of `BaseView.attach`, the order in which we get results from `BaseView.getView` matters.
Indeed, if for example `childView1` is returned before `childViewOfChildView1`, the former will be `attach(ed)OrRender(ed)` immediately, which eventually will lead to another `BaseView.attach` call, where the array of subviews will only be:

```
[ childViewOfChildView1 ]
```

* The moment we then pass this check https://github.com/rendrjs/rendr/blob/master/shared/base/view.js#L450
the view is going to be rendered twice, because `attachOrRender` is going to be reached and executed both by this current event loop and by the one that started the first BasView.attach, when `BaseView.getView` returns a result for `childViewOfChildView1`.

I hope I made this clear enough. If that's not the case, please let me know and I'll try to provide a better explanation. This issue gave me a bit of headache :)

A fix for this would be as easy as just adding this

```
if(!$el.data('view-attached'))
```
before this line: https://github.com/rendrjs/rendr/blob/master/shared/base/view.js#L331
Or rather, we could probably mark and filter out subviews from the initial array, but that would definitely take more time and work.

Does this all make sense? Please let me know if I'm just missing something instead.

**Comment by ingiulio** [comment]
On a second thought about the fix, that should probably go even before the view is created, so before this line: https://github.com/rendrjs/rendr/blob/master/shared/base/view.js#L460
I'll just file a PR

**Comment by ingiulio** [comment]
Fixed by #463
